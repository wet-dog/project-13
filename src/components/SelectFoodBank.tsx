import { FormControl, Select } from "native-base";
import React, { useEffect, useState } from "react";

import { collection} from "firebase/firestore";
import { useCollectionData } from 'react-firebase-hooks/firestore';

import { db } from "../utils/firebase";

type SelectFoodBankProps = {
  setBank: (bank: string) => void
}

function SelectFoodBank(props: SelectFoodBankProps) {

  const banksRef = collection(db, "foodBank");

  const [banks, loading, error, snapshot] = useCollectionData(banksRef);

  return (
    <FormControl w="3/4" maxW="300" isRequired isInvalid>
      <FormControl.Label>Select Food Bank</FormControl.Label>
      <Select  
        onValueChange={(value) => props.setBank(value)} 
        minWidth="200" 
        accessibilityLabel="Choose Food Bank" 
        placeholder="Choose Food Bank" 
        _selectedItem={{bg: "teal.600",}} 
        mt="1"
      >
      { 
        banks 
        ? banks.map(bank => <Select.Item label={ String(bank.bankName) } value={ String(bank.bankName) } />)
        : <Select.Item label="" value="" />
      }
      </Select>
      <FormControl.ErrorMessage>
        Please make a selection!
      </FormControl.ErrorMessage>
    </FormControl>
  );
}

export default SelectFoodBank;
