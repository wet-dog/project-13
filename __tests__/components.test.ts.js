import renderer from "react-test-renderer";
import React from "react";
import SelectFoodBank from "../src/components/SelectFoodBank";
import { NativeBaseProvider } from "native-base";

describe("SelectFoodBank component test", () => {
    it('SelectFoodBank component test', () => {
        const component = renderer.create(<NativeBaseProvider><SelectFoodBank setBank={() => {}}/></NativeBaseProvider>).toJSON();
        expect(component).toMatchSnapshot();
    });
});
