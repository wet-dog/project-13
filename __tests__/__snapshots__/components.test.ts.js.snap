// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFoodBank component test SelectFoodBank component test 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
